src/software/ai/hl/stp/tactic/crease_defender/crease_defender_tactic_test.py

```diff
@@ -0,0 +1,367 @@
+                ball_velocity=ball_initial_velocity,
+            ),
+        )
+
+        # These aren't necessary for this test, but this is just an example
+        # of how to send commands to the simulator.
+        #
+        # NOTE: The gamecontroller responses are automatically handled by
+        # the gamecontroller context manager class
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.STOP, team=Team.UNKNOWN
+        )
+        simulated_test_runner.gamecontroller.send_gc_command(
+            gc_command=Command.Type.FORCE_START, team=Team.BLUE
+        )
+
+        # Setup Tactic
+        params = AssignedTacticPlayControlParams()
+
+        params.assigned_tactics[0].crease_defender.CopyFrom(
+            CreaseDefenderTactic(
+                enemy_threat_origin=tbots_cpp.createPointProto(ball_initial_pos),
+                crease_defender_alignment=2,
+                max_allowed_speed_mode=MaxAllowedSpeedMode.PHYSICAL_LIMIT,
+                ball_steal_mode=BallStealMode.STEAL,
+            )
+        )
+
+        simulated_test_runner.blue_full_system_proto_unix_io.send_proto(
+            AssignedTacticPlayControlParams, params
+        )
+
+        # Setup no tactics on the enemy side
+        params = AssignedTacticPlayControlParams()
+        simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+            AssignedTacticPlayControlParams, params
+        )
+
+    # Always Validation
+    always_validation_sequence_set = [
+        [
+            RobotNeverEntersRegion(
+                regions=[
+                    tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea(),
+                    tbots_cpp.Field.createSSLDivisionBField().enemyHalf(),
+                    tbots_cpp.Circle(ball_initial_pos, 1),
+                ]
+            ),
+            NeverExcessivelyDribbles(),
+        ]
+    ]
+
+    # Eventually Validation
+    eventually_validation_sequence_set = [
+        [
+            # Crease defender should be near the defense area
+            RobotEventuallyEntersRegion(
+                regions=[
+                    tbots_cpp.Field.createSSLDivisionBField()
+                    .friendlyDefenseArea()
+                    .expand(0.25)
+                ]
+            ),
+        ]
+    ]
+
+    simulated_test_runner.run_test(
+        setup=setup,
+        inv_eventually_validation_sequence_set=eventually_validation_sequence_set,
+        inv_always_validation_sequence_set=always_validation_sequence_set,
+        ag_eventually_validation_sequence_set=eventually_validation_sequence_set,
+        ag_always_validation_sequence_set=always_validation_sequence_set,
+        test_timeout_s=4,
+    )
+
+
+friendlyDefenseAreaFrontCenter = tbots_cpp.Point(
+    tbots_cpp.Field.createSSLDivisionBField()
+    .friendlyDefenseArea()
+    .posXPosYCorner()
+    .x(),
+    tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea().centre().y(),
+)
```

Comment:
this is just `tbots_cpp.Field.createSSLDivisionBField().friendlyDefenseArea().centre()` btw

Reply:
The point I want is the center of the frontal segment of the defence area, not necessarily the exact center of the defense box.